### PR TITLE
Add UserPermission

### PR DESCRIPTION
**Pull request by CatSue**
## Fixes

  -
  -

## Proposed Changes

  - Add UserPermission in metadataType
  - Manage elements in PS config that are not in packagedXml

### CODE REVIEW

File: src/commands/essentials/permissionset/generate.ts [hunk 1]
```diff
@@ -329,6 +329,22 @@ export default class PermissionSetGenerate extends Command {
...
     public filterPackageXmlTypes(packageXmlTypes: any, packageXMLTypesAll: any, permissionSetDefinition: any) {
         let permissionSetsXmlElement = '';
+
+        let packageXmlTypesArray = [];
+        for (const packageXmlTypeElement of packageXmlTypes) {
+            packageXmlTypesArray.push(packageXmlTypeElement.name[0]);
+        }
+        // Manage elements that are in permissionSetDefinition JSON and not in packageXmlTypes
+        for (const permissionSetElement of permissionSetDefinition.packageXMLTypeList) {
+            if (!packageXmlTypesArray.includes(permissionSetElement.typeName) && packageXMLTypesAll.includes(permissionSetElement.typeName) ) {
```
Comment (nvuillam) [inline src/commands/essentials/permissionset/generate.ts]: That's why you have a member with (.*) in the generate PS xml file  ...
It's possible that the solution is not to add virtual stuff in packageXmlTypes, but to add the element in packageXmlTypesArray

File: src/common/metadata-utils/index.ts [hunk 2]
```diff
@@ -59,11 +59,12 @@ class MetadataUtils {
       CompactLayout: { sobjectRelated: true },
       CustomField: { sobjectRelated: true, permissionSetTypeName: 'fieldPermissions', permissionSetMemberName: 'field' },
       CustomObject: { sobjectRelated: true, permissionSetTypeName: 'objectPermissions', permissionSetMemberName: 'object' },
+      FieldSet: { sobjectRelated: true },
       ListView: { sobjectRelated: true },
       RecordType: { sobjectRelated: true, permissionSetTypeName: 'recordTypeVisibilities', permissionSetMemberName: 'recordType' },
-      WebLink: { sobjectRelated: true },
+      UserPermission: { sobjectRelated: true, permissionSetTypeName: 'userPermissions', permissionSetMemberName: 'name' },
```
Comment (nvuillam) [inline src/common/metadata-utils/index.ts]: are you sure UserPermission is related to a SObject ?

File: src/commands/essentials/permissionset/generate.ts (hunk 1 above)
Comment (CatSue) [reply src/commands/essentials/permissionset/generate.ts]: Add check on includedFilterList and additionalElements

File: src/common/metadata-utils/index.ts (hunk 2 above)
Comment (CatSue) [reply src/common/metadata-utils/index.ts]: Replaced by sobjectRelated: false

**Pull request merged**